src/server/zset_family.cc

```diff
@@ -222,6 +222,65 @@ OpResult<DbSlice::ItAndUpdater> FindZEntry(const ZParams& zparams, const OpArgs&
   return DbSlice::ItAndUpdater{add_res.it, add_res.exp_it, std::move(add_res.post_updater)};
 }
 
+using RandomPick = std::size_t;
+
+class PicksGenerator {
+ public:
+  virtual RandomPick Generate() = 0;
+  virtual ~PicksGenerator() = default;
+};
+
+class NonUniquePicksGenerator : public PicksGenerator {
```

Comment:
This classes are generic generators
I think that total_size_ can be renamed to max_range type uint32_t
also return value from Generate can be changed to uint32_t